testapi.pm

```diff
@@ -1144,22 +1144,30 @@ sub release_key {
 
 =head2 send_key_until_needlematch
 
-  send_key_until_needlematch($tag, $key [, $counter, $timeout]);
+  send_key_until_needlematch($tag, $key [, [$counter] | [counter => $counter], [$timeout] | [timeout => $timeout], [no_wait => $no_wait]]);
+
 
 Send specific key until needle with C<$tag> is not matched or C<$counter> is 0.
-C<$tag> can be string or C<ARRAYREF> (C<['tag1', 'tag2']>)
-Default counter is 20 steps, default timeout is 1s
+C<$tag> can be string or C<ARRAYREF> (C<['tag1', 'tag2']>).
+Default counter is 20 steps, default timeout is 1s.
+
+Specify C<$no_wait> to run the screen check as fast as possible that is
+possibly more than once per second which is default. Select this to check a
+screen which can change in a range faster than 1-2 seconds not to miss the
+screen to check for.
 
 Throws C<FailedNeedle> exception if needle is not matched until C<$counter> is 0.
 
 =cut
 
 sub send_key_until_needlematch {
-    my ($tag, $key, $counter, $timeout) = @_;
+    my $tag     = shift;
+    my $key     = shift;
+    my $counter = looks_like_number($_[0]) ? shift : 20;
+    my $timeout = looks_like_number($_[0]) ? shift : 1;
+    my %args    = (counter => $counter, timeout => $timeout, @_);
 
```

Comment:
I like the approach, but the changes made to the parsing of the method arguments could lead it to set a timeout of 1  if the method is called as `send_key_until_needlematch('mytag', 'mykey', counter => 5, 34)` (which granted, is not in use right now, but could be valid per the comment in L1147). I would recommend to either change the comment, or make the argument parsing more thorough.

Reply:
My suggestion for this:

```
my $counter = ($_[0] eq 'counter') ? $_[1] : looks_like_number($_[0]) ? $_[0] : 20;
$counter == $_[0] ? splice(@_, 0, 1) : splice(@_, 0, 2);
my $timeout = ($_[0] eq 'timeout') ? $_[1] : looks_like_number($_[0]) ? $_[0] : 1;
$timeout == $_[0] ? splice(@_, 0, 1) : splice(@_, 0, 2);
my %args    = (timeout => $timeout, @_);
```

Yeah, I know. Not too pretty. Perhaps will be easier to read with if-else blocks.